syntax = "proto3";

package hello;

// Definici√≥n del servicio CRUD
service UserService {
  rpc CreateUser (UserRequest) returns (UserReply);
  rpc GetUser (UserIdRequest) returns (UserReply);
  rpc UpdateUser (UserRequest) returns (UserReply);
  rpc DeleteUser (UserIdRequest) returns (UserReply);
}

// Mensaje para crear/actualizar usuarios
message UserRequest {
  int32 id = 1;
  string name = 2;
  string email = 3;
}

// Mensaje para buscar/eliminar por ID
message UserIdRequest {
  int32 id = 1;
}

// Respuesta del servidor
message UserReply {
  int32 id = 1;
  string name = 2;
  string email = 3;
  string message = 4;
}
